src/test/java/org/hydev/mcpm/client/database/SearchInteractorTest.java

```diff
@@ -0,0 +1,76 @@
+package org.hydev.mcpm.client.database;
+
+import org.hydev.mcpm.client.database.fetcher.LocalDatabaseFetcher;
+import org.hydev.mcpm.client.database.inputs.SearchPackagesInput;
+import org.hydev.mcpm.client.database.results.SearchPackagesResult;
+import org.hydev.mcpm.client.database.searchusecase.SearchInteractor;
+import org.junit.jupiter.api.Test;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Test suite for the SearchInteractor class.
+ *
+ * @author Jerry Zhu (<a href="https://github.com/jerryzhu509">...</a>)
+ */
+public class SearchInteractorTest {
+    private final URI host = URI.create("http://mcpm.hydev.org");
+    private final LocalDatabaseFetcher fetcher = new LocalDatabaseFetcher(host);
+    private final SearchInteractor database = new SearchInteractor(fetcher);
+
+    /**
+     * Helper method for formatting the output as a string.
+     *
+     * @param result Search result.
+     * @param delim Delimiter separating names.
+     * @return Formatted names of plugins as a string.
+     */
+    private String formatStr(SearchPackagesResult result, String delim) {
+        return result
+                .plugins()
+                .stream()
+                .map(x -> x.versions().stream().findFirst().map(value -> value.meta().name()))
```

Comment:
I'm concerned about the message chain here; If it works, that's good, but when we've got the time it would be helpful to refactor and add more documentation.

Reply:
This is just a private helper method in the Test to format the output, and is not needed anywhere else.

Reply:
This looks sane to me.